pandas/tests/groupby/test_transform.py

```diff
@@ -782,3 +782,23 @@ def test_any_all_np_func(func):
 
     res = df.groupby('key')['val'].transform(func)
     tm.assert_series_equal(res, exp)
+
+
+def test_transform_with_all_nan():
+    # GH 21624
+    df = DataFrame({'groups': [np.nan, np.nan, np.nan],
+                    'values': [1, 2, 3]})
+
+    grouped = df.groupby('groups')
+    result = grouped['values'].transform('sum')
+    expected = Series([np.nan, np.nan, np.nan], name='values')
+    tm.assert_series_equal(result, expected)
+
+    # Make sure the standard case still works too
+    df = DataFrame({'groups': [np.nan, 'A', 'A', 'B', 'B'],
```

Comment:
Since this follows the same pattern as the code above it would be preferable to parametrize it instead of re-coding

Reply:
This is done in the latest update